lstchain/image/modifier.py

```diff
@@ -105,10 +121,235 @@ def random_psf_smearer(image, fraction, indices, indptr):
+    
+
+    # Pulse integration window width of the (biased) extractor for showers:
+    shower_extractor_window_width = config[config['image_extractor']]['window_width']
+
+    # Pulse integration window width for the pedestal estimation:
+    pedestal_extractor_window_width = config['LSTCalibrationCalculator']\
+        ['FixedWindowSum']['window_width']
+
+    # MC pedestals integrated with the unbiased pedestal extractor
+    mc_ped_charges = []
+    # MC pedestals integrated with the biased shower extractor
+    mc_ped_charges_biased = []
+
+    for event in mc_reader:
+        if tel_id not in event.trigger.tels_with_trigger:
+            continue
+        # Extract the signals as we do for pedestals (unbiased fixed window
+        # extractor):
+        pedestal_calibrator(event)
+        charges = event.dl1.tel[tel_id].image
+
+        # True number of pe's from Cherenkov photons (to identify noise-only pixels)
+        true_image = event.simulation.tel[tel_id].true_image
+        mc_ped_charges.append(charges[true_image == 0])
+
+        # Now extract the signal as we would do for shower events (usually
+        # with a biased extractor, e.g. LocalPeakWindowSum):
+        shower_calibrator(event)
+        charges_biased = event.dl1.tel[tel_id].image
+        mc_ped_charges_biased.append(charges_biased[true_image == 0])
+
+    # All pixels behave (for now) in the same way in MC, just put them together
+    mc_ped_charges = np.concatenate(mc_ped_charges)
+    mc_ped_charges_biased = np.concatenate(mc_ped_charges_biased)
+
+    mcq = np.histogram(mc_ped_charges_biased, bins=qbins, range=qrange,
+                       density=True)
+    # Find the peak of the pedestal biased charge distribution of MC. Use
+    # an interpolated version of the histogram, for robustness:
+    func = interp1d(0.5*(mcq[1][1:]+mcq[1][:-1]), mcq[0],
+                    kind='quadratic', fill_value='extrapolate')
+    xx = np.linspace(qrange[0], qrange[1], 100*qbins)
+    mode_mc = xx[np.argmax(func(xx))]
+
+    mc_unbiased_std_ped_pe = np.std(mc_ped_charges)
+
+    # Find the additional noise (in data w.r.t. MC) for the unbiased extractor,
+    # and scale it to the width of the window for integration of shower images.
+    # The idea is that when a strong signal is present, the biased extractor
+    # will integrate around it, and the additional noise is unbiased because
+    # it won't modify the integration range.
+    extra_noise_in_bright_pixels = \
+        ((data_median_std_ped_pe ** 2 - mc_unbiased_std_ped_pe ** 2) *
+         shower_extractor_window_width / pedestal_extractor_window_width)
+
+    # Just in case, makes sure we just add noise if the MC noise is smaller
+    # than the real data's:
+    extra_noise_in_bright_pixels = max(0., extra_noise_in_bright_pixels)
+
+    bias = mode_data - mode_mc
+    extra_bias_in_dim_pixels = max(bias, 0)
+
+    # differences of values to peak charge:
+    dq = data_ped_charges[all_good].flatten() - mode_data
```

Comment:
what is `dq`? `charge_difference`? maybe explicitly naming it?

Reply:
Would make the later expressions clumsy. And it is clearly explained, right? Differences of values to peak charge (dq stands for delta_charge). Also, they are used very close to their initialization so it is quite clear I think.